convert2rhel/logger.py

```diff
@@ -88,6 +90,35 @@ def initialize_logger(log_name, log_dir=LOG_DIR):
     logger.addHandler(handler)
 
 
+def archive_old_logger_files():
+    """Archive the old log files to not mess with multiple runs outputs.
+    Every time a new run begins, this method will be called to archive the previous logs
+    if there is a `convert2rhel.log` file there, it will be archived using
+    the same name for the log file, but having an appended timestamp to it.
+
+    For example:
+        /var/log/convert2rhel/archive/convert2rhel-1635162445070567607.log
+        /var/log/convert2rhel/archive/convert2rhel-1635162478219820043.log
+
+    This way, the user can track the logs for each run individually based on the timestamp.
+    """
+
+    current_log_file = "%s/convert2rhel.log" % LOG_DIR
+
+    if os.path.exists(current_log_file):
+        stat = os.stat(current_log_file)
+
+        # Get the last time the file was modified as nanoseconds
+        last_modified_at = stat.st_mtime_ns
+
+        # Create the directory if it don't exist
+        if not os.path.isdir(ARCHIVE_LOG_DIR):
```

Comment:
This is a directory check but the logic is really about existence.  Maybe this should be `os.path.exists()` instead?

Additionally, maybe we want to check if it is a directory afterwards and fail with an exception that we can catch at the higher level to emit a nice error message (Like `"%s exists but is not a directory.  Old log files not archived." % ARCHIVE_LOG_DIR`)  (We can also catch whatever exceptions shutil.copyfile  emits rather than checking whether we're using a directory explicitly.  I'm not sure, though, how many other things copyfile could fail on that we'd have to look through.)

Lastly, just a future note:  Using `os.makedirs()` like this is sometimes insecure.  As long as we're hardcoding the path to something that isn't writable by other users, we should be okay but we'd need to worry about it if we make this user settable (as the user could set it to something like `/tmp/convert2rhel/` which would lead to symlink attacks)

Reply:
Addtion to the insecure note.... the way we're constructing the path to the logfile means that if either the `log_dir` or `log_name` become user settable we'd have issues.  We may want to write some mitigating code in logging.py as the log_dir and log_name are both passed into this set of functions.  We're relying on main.py to hardcode the values.  Someone changing how the functions are called from main.py may not realize the security implications that raises here.

Reply:
Great catch, @abadger. Just to be sure on that exception you mentioned. At this point of code, we would have something like this? 

```python

# In logger.py:
if not os.path.exists(ARCHIVE_LOG_DIR):
    ...
else:
    raise FileNotFoundError
```

And in an earlier stage, let's say, on `logger.py` we would have something like 

```python
def initialize_logger(log_name, log_dir=LOG_DIR):
    
    try:
        archive_old_logger_files(log_name=log_name)
    except FileNotFoundError:
        ...

    setup_logger(log_name=log_name, log_dir=log_dir)
 ```
 
 Or do you have any better ideas?